core/src/com/unciv/ui/components/widgets/HealthBarV0.kt

```diff
@@ -0,0 +1,317 @@
+package com.unciv.ui.components.widgets
+
+import com.badlogic.gdx.graphics.Color
+import com.badlogic.gdx.graphics.g2d.Batch
+import com.badlogic.gdx.math.Interpolation
+import com.badlogic.gdx.math.Rectangle
+import com.badlogic.gdx.scenes.scene2d.actions.FloatAction
+import com.badlogic.gdx.scenes.scene2d.ui.Label
+import com.badlogic.gdx.scenes.scene2d.ui.Widget
+import com.badlogic.gdx.scenes.scene2d.utils.Drawable
+import com.badlogic.gdx.utils.Align
+import com.unciv.logic.city.City
+import com.unciv.models.ruleset.INonPerpetualConstruction
+import com.unciv.ui.components.extensions.darken
+import com.unciv.ui.components.fonts.Fonts
+import com.unciv.ui.components.tilegroups.CityButton
+import com.unciv.ui.images.ImageGetter
+import com.unciv.ui.screens.basescreen.BaseScreen
+import kotlin.math.PI
+import kotlin.math.cos
+
+@Suppress("MemberVisibilityCanBePrivate", "unused")
+
+//TODO Since HealthBar instances can share style instances with the same values, use common ones.
+//      - skin, dedicated cache, enum, or companion vals?
+
+/**
+ *  A reusable Widget to draw progress bars, health bars and the like.
+ */
+
+class HealthBarV0(
+    val minValue: Float,
+    val maxValue: Float,
+    segmentCount: Int = 2,
+    isVertical: Boolean = false,
+    var style: HealthBarStyle = HealthBarStyle(isVertical)
+) : Widget() {
+    /**
+     *  A reusable Widget to draw progress bars, health bars and the like.
+     *  Convenience constructor using Int limits, and default constructor.
+     *
+     *  Note that a default instance without any further configuration will draw nothing and have zero pref size.
+     */
+    constructor(
+        minValue: Int = 0,
```

Comment:
Let's not even allow minValue other than zero - that's nonsense for a healthbar.

Reply:
That sounds like an entirely different set of problems - that bar extends *from the center* towards the other side
These are two very different problem spaces